File: Easy Questions/Best-Time-To-Buy-Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int price = prices[0];
        int profit = 0;
        
        for(int i=1; i< prices.size(); ++i){
            profit = max(profit, prices[i]- price);
            price = min(prices[i], price);
            
        } 
        return profit;
    } 
};

File: Easy Questions/Best-Time-To-Buy-Stock.rs
use std::cmp::{max, min};

impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut low = i32::MAX; 
        let mut res = 0;
        for i in prices {
            res = max(res, i - low);
            low = min(low, i);
        }
        res
    }
}
